packages/internals/src/utils/fs-functional.ts

```diff
@@ -0,0 +1,68 @@
+import { pipe } from 'fp-ts/lib/function'
+import * as T from 'fp-ts/lib/Task'
+import * as TE from 'fp-ts/lib/TaskEither'
+import fs from 'fs'
+import globby from 'globby'
+import path from 'path'
+
+export const createDirIfNotExists = (dir: string) =>
+  pipe(
+    TE.tryCatch(
+      // Note: { recursive: true } prevents EEEXIST error codes when the directory already exists
+      () => fs.promises.mkdir(dir, { recursive: true }),
+      createTaggedSystemError('fs-create-dir', { dir }),
+    ),
+  )
+
+export const writeFile = ({ path, content }: { path: string; content: string }) =>
+  pipe(
+    TE.tryCatch(
+      () => fs.promises.writeFile(path, content, { encoding: 'utf-8' }),
+      createTaggedSystemError('fs-write-file', { path, content }),
+    ),
+  )
+
+/**
+ * Note to future self: in Node.js, `removeDir` and `removeFile` can both be implemented with a single `fs.promises.rm` call.
+ */
+
+export const removeDir = (dir: string) =>
+  pipe(
+    TE.tryCatch(() => fs.promises.rmdir(dir, { recursive: true }), createTaggedSystemError('fs-remove-dir', { dir })),
+  )
+
+export const removeFile = (filePath: string) =>
+  pipe(TE.tryCatch(() => fs.promises.unlink(filePath), createTaggedSystemError('fs-remove-file', { filePath })))
+
+/**
+ * Removes all backslashes from a possibly Windows path string, which is necessary for globby to work on Windows.
+ */
+const normalizePossiblyWindowsDir = (dir: string) => dir.replace(/\\/g, '/')
```

Comment:
A backslash is a valid filename character on Linux and macOS, so this will break in that edge case. I'd replace `path.sep` instead of `\\` here, but keeping an explicit backslash is also an option if you wrap it into `process.platform === 'win32'` check.